components/script/dom/document.rs

```diff
@@ -574,6 +578,28 @@ impl Document {
         }
     }
 
+    /// https://html.spec.whatwg.org/multipage/#scroll-to-the-fragment-identifier
+    pub fn check_and_scroll_fragment(&self, fragment: &str) {
+        let body = self.GetBody();
+        let target = self.find_fragment_node(fragment);
+
+        // Step 1
+        self.set_target_element(target.r());
+
+        let target = target.r().or_else(||
+                                        // Step 2
+                                        if fragment.is_empty() {
+                                            body.r().map(|b| b.upcast::<Element>())
```

Comment:
So you mean `scroll_fragment_point` would spit out an error if `target` is a `Document` element? This means that you'll need to check whether `fragment == "top"` as well, since `find_fragment_node` will return a `Document`.